manager/integration/tests/test_node.py

```diff
@@ -0,0 +1,375 @@
+
+    # reduce StorageReserved of each default disk
+    nodes = client.list_node()
+    for node in nodes:
+        disks = node["disks"]
+        update_disks = get_update_disks(disks)
+        for disk in update_disks:
+            disk["storageReserved"] = 0
+        node.diskUpdate(disks=update_disks)
+
+    # check volume status
+    volume = common.wait_for_volume_detached(client, vol_name)
+    conditions = volume["conditions"]
+    assert conditions["scheduled"]["status"]
+    assert volume["state"] == "detached"
+    assert volume["created"] != ""
+
+    volume.attach(hostId=lht_hostId)
+    volume = common.wait_for_volume_healthy(client, vol_name)
+    nodes = client.list_node()
+    node_hosts = []
+    for node in nodes:
+        node_hosts.append(node["name"])
+    # check all replica should be scheduled to default disk
+    for replica in volume["replicas"]:
+        id = replica["hostId"]
+        assert id != ""
+        assert replica["running"]
+        expect_disk = expect_node_disk[id]
+        for key, disk in expect_disk.iteritems():
+            assert replica["diskID"] == key
+            assert disk["path"] in replica["dataPath"]
+            break
+        node_hosts = filter(lambda x: x != id, node_hosts)
+    assert len(node_hosts) == 0
+
+    # clean volume and disk
+    clean_volume(client, vol_name)
+
+    node = client.by_id_node(lht_hostId)
+    disks = node["disks"]
+    disk = disks[unexpected_disk["fsid"]]
+    disk["allowScheduling"] = False
+    update_disks = get_update_disks(disks)
+    node = node.diskUpdate(disks=update_disks)
+    # wait for node controller to update disk status
+    time.sleep(WAIT_FOR_NODE_STATUS)
+    disks = node["disks"]
+    disks.pop(unexpected_disk["fsid"])
+    update_disks = get_update_disks(disks)
+    node.diskUpdate(disks=update_disks)
+    clean_disk(client, 'vol-small')
+
+
+def test_node_controller(clients):  # NOQA
+    # get a random client
+    for host_id, client in clients.iteritems():
+        break
+
+    lht_hostId = get_self_host_id()
+    nodes = client.list_node()
+
+    # create a volume and test update StorageScheduled of each node
+    vol_name = common.generate_volume_name()
+    volume = create_volume(client, vol_name, str(1*Gi), lht_hostId, len(nodes))
+    replicas = volume["replicas"]
+
+    # wait for node controller to update disk status
+    time.sleep(WAIT_FOR_NODE_STATUS)
+    nodes = client.list_node()
+    for node in nodes:
+        disks = node["disks"]
+        for replica in replicas:
+            if replica["hostId"] == node["name"]:
+                disk = disks[replica["diskID"]]
+                assert disk["storageScheduled"] == 1*Gi
+                break
+
+    # clean volumes
+    clean_volume(client, vol_name)
+
+    # write specified byte data into disk
+    if os.path.exists(TEST_FILE):
+        os.remove(TEST_FILE)
+    cmd = ['dd', 'if=/dev/zero', 'of=' + TEST_FILE, 'bs=1M', 'count=1024']
```

Comment:
This test file is way too big. Should we able to detect e.g. 10M? That should be enough.